src/cloud-api-adaptor/podvm/hack/smoke_test.sh

```diff
@@ -197,3 +353,35 @@ if ! $KATACTL connect --server-address "unix://${SOCK}" --cmd DestroySandbox; th
 	echo "::error:: Failed to DestroySandbox"
 	exit 1;
 fi
+
+if [ "$TEST_MODE" = "scratch-space" ]; then
+	if ! check_scratch_disk_mount "${VM_NAME}"; then
+		echo "::error:: Scratch disk not found"
+		exit 1
+	fi
+	echo "::debug:: Scratch disk check passed for $TEST_MODE mode"
+fi
+
+if [ "$TEST_MODE" = "encrypted-scratch-space" ]; then
+	if ! check_encrypted_disk_mount "${VM_NAME}"; then
+		echo "::error:: Encrypted disk not found"
+		exit 1
+	fi	
+
+	# Copy a unique string to the VM under /run/kata-containers/image using virsh qemu-agent-command
+	# Search for the string in the VM image from the host (only for scratch space modes)
+	echo "::debug:: Testing encrypted disk space by writing a unique string"
+	UNIQUE_STRING="smoketest-podvm-123456"
+
+	sudo virsh qemu-agent-command "${VM_NAME}" \
+	 '{"execute": "guest-exec", "arguments": { "path": "/bin/bash", "arg": [ "-c", "echo smoketest-podvm-123456 > /run/kata-containers/image/smoketest.txt" ], "capture-output": false }}'
```

Comment:
we can run sync to make sure the string is actually written to the disk. the best would probably to umount the device, but that should do it. 

```suggestion
	 '{"execute": "guest-exec", "arguments": { "path": "/bin/bash", "arg": [ "-c", "echo smoketest-podvm-123456 > /run/kata-containers/image/smoketest.txt && sync" ], "capture-output": false }}'
```